sporadic failures with travis "g++: internal compiler error: Killed (program cc1plus)"

Every now and then a build will fail on travis during compilation with the error message:

g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.6/README.Bugs> for instructions.
make[2]: *** [CMakeFiles/run_elm_unittests.dir/modules/elm/layers/test/triangulation_unittests.cpp.o] Error 4
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/run_elm_unittests.dir/all] Error 2
make: *** [all] Error 2

Sometimes repeating the build once or twice resolves this.

This seems to be due to g++ hitting the memory limits of the 3 GB Travis VM.

I've tried adding "--param ggc-min-expand=3 --param ggc-min-heapsize=5120" to the CXX_FLAGS, to trigger garbage collection more frequently, but don't think it helped much. More frequent triggering by increasing the value for ggc-min-expand slows down the process so much it leads to the build timing out.

The issue is addressed in [Travis CI issue #1972](https://github.com/travis-ci/travis-ci/issues/1972)

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,7 +21,7 @@ env:
     - TRAVIS_PYTHON_VERSION="2.7"
     # ELM build parameters
     - BUILD_TYPE=Coverage
-    - BUILD_SHARED_LIBS=OFF
+    - BUILD_SHARED_LIBS=ON
     - secure: "fvYVNecz00QbdEz9sWc5bUjaCby7fsrfKzkFCTzsYmI2PzRB1CMKDpIxjxI77z/KixFeiZkQ2IAQck13EvWCnqEccBjt/MmrmJGWBLwpK1yAaJe4opFICQcUwvmG8GN0/ppi7g+QD9Svu+mrMEsVBoQm9nX+NdvMAMSMiX7ZXFg="
 
 before_install:
@@ -36,7 +36,7 @@ before_install:
   - travis_retry wget --quiet https://googletest.googlecode.com/files/gtest-1.7.0.zip
   - unzip -qq gtest-1.7.0.zip
   - cd gtest-1.7.0
-  - cmake . > /dev/null
+  - cmake -DBUILD_SHARED_LIBS=$BUILD_SHARED_LIBS . > /dev/null
   - make > /dev/null
   - cd $DEPS_DIR
   

